OKlib.js

```diff
@@ -824,416 +767,378 @@ registerPlugin({
     }
 
     /**
-     * Concatenates two Arrays
-     * @param  {Object[]} array
-     * @param  {Object[]} elements
-     * @return {Object[]} Returns a new Array containing all Elements of the two given Arrays
+     * concatenates two arrays
+     * @param {Object[]} array
+     * @param {Object[]} elements
+     * @return {Object[]} returns a new array containing all elements of the two given arrays
      */
     function arrayCombineArrays(array, elements) {
         array = arrayCreateArray(array);
         elements = arrayCreateArray(elements);
         let result = [];
-        for (let i = 0; i < array.length; i++) {
+        for (let i in array) {
```

Comment:
change this to for..of or revert to normal for loop, because for..in is not safe for array operation (might not iterate in the right order)